src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,295 @@
+            System.out.println("\tNow you have " + Integer.toString(numberOfTasks) + " task(s) in the list! 凸(￣ヘ￣)");
+            System.out.println("");
+            printDivider();
+        }
+    }
+
+    public static void removeTask(int index) {
+        try {
+            Task removedTask = tasks.get(index);
+            tasks.remove(index);
+            numberOfTasks--;
+
+            printDivider();
+            System.out.println("\tNoted! (๑˃ᴗ˂)ﻭ I've removed this task:");
+            System.out.print("\t  ");
+            System.out.println(removedTask);
+            System.out.println("\tNow you have " + numberOfTasks + " task(s) in the list! 凸(￣ヘ￣)");
+            System.out.println("");
+            printDivider();
+        } catch (IndexOutOfBoundsException e) {
+            printExceptionMessage(EXCEPTION_3);
+        }
+    }
+
+    public static void printList() {
+        printDivider();
+        System.out.println("\tHere are the tasks in your list:");
+        for (int i = 0; i < numberOfTasks; i++) {
+            System.out.print("\t" + (i + 1) + ".");
+            System.out.println(tasks.get(i));
+        }
+        System.out.println();
+        printDivider();
+    }
+
+    public static void markAsDone(String index, boolean isFile) {
+        try {
+            Task currentTask = tasks.get(Integer.parseInt(index) - 1);
+            currentTask.markAsDone();
+            rewriteFile();
+
+            if (!isFile) {
+                printDivider();
+                System.out.println("\tNice! (〃＾▽＾〃) I've marked this task as done:");
+                System.out.println("\t  [" + currentTask.getStatusIcon() + "] " + currentTask.description);
+                System.out.println("\tWell done completing your task, friend! (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧");
+                System.out.println("");
+                printDivider();
+            }
+        } catch (NullPointerException e) {
+            printExceptionMessage(EXCEPTION_3);
+        } catch (IOException e) {
+            printExceptionMessage(EXCEPTION_4);
+        }
+    }
+
+    public static void markAsUndone(String index, boolean isFile) {
+        try {
+            Task currentTask = tasks.get(Integer.parseInt(index) - 1);
+            currentTask.markAsUndone();
+            rewriteFile();
+
+            if (!isFile) {
+                printDivider();
+                System.out.println("\tOK, I've marked this task as not done yet:");
+                System.out.println("\t  [" + currentTask.getStatusIcon() + "] " + currentTask.description);
+                System.out.println("\tYou better stop procrastinating, friend.. (╥﹏╥)");
+                System.out.println("");
+                printDivider();
+            }
+        } catch (NullPointerException e) {
+            printExceptionMessage(EXCEPTION_3);
+        } catch (IOException e) {
+            printExceptionMessage(EXCEPTION_4);
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
Try to structure code logically, your code should read like a story and a way to do this is by ordering the methods in their order of being called. Since main is the first function it can be listed first and other methods can be ordered in the order that they are being called in the program flow to improve readability.

Code Quality Guideline: Structure Code Logically